components/brave_ads/browser/BUILD.gn

```diff
@@ -42,28 +42,40 @@ source_set("browser") {
       "ads_service_impl.h",
       "background_helper.cc",
       "background_helper.h",
-      "background_helper_android.cc",
```

Comment:
These files are built only for their respective OS's, we should not need to wrap in these conditions. Did you try as I know windows, macOS and Linux work in this way when building brave-browser?

Reply:
BUILD.gn has been reverted.